Features/prot 66 connect onboarding fields to firestore

chelseaL01 opened the pull request.

chelseaL01 committed:
Add Dialogs and Recycler Views

Dialogs and Recycler Views are for the following: Allergy, Cancer, Diabetes, Surgery, OtherMedicalConditions.

The Allergy Dialog is functional: Users can enter allergy name, diagnosis date, and doctor who diagnosed. If the dialog is completed correctly, the inputs will be shown on the allergy recycler view.

chelseaL01 committed:
Add dialog and recycler view functionality

The dialog and recycler views for the following information are now functional: Cancer, Diabetes, Surgery, and Other Medical Conditions

Created variables for the other onboarding fields

chelseaL01 committed:
Add Onboarding Fields to Firestore

Also check to see if fields like phone numbers, date of birth, and height inches are the correct format.

The required fields include general section, emergency contact section, and the height from the medical section.

The Cancer, Diabetes, Surgery, Other Conditions, and Allergies fields in the medical section are not required to submit the onboarding form

chelseaL01 committed:
Add onboarding completed flag + General Health Information page

Add Onboarding Completed Flag and show the following onboarding fields on General Health Information page:

General Health - phone number, height, address
Emergency Contact - name, phone number, email

chelseaL01 committed:
Update PatientDashboardActivity.java

chelseaL01 committed:
Add Recycler Views to General Information page

Update the general Information page to show any Surgeries, Cancer, Diabetes, Allergies, or Other Medical Conditions data stored on Firestore from the onboarding form.

chelseaL01 committed:
Merge branch 'main' of https://github.com/nickdlc/Protego into features/PROT-66-Connect-Onboarding-Fields-to-Firestore

 Conflicts:
	client/.idea/misc.xml
	client/app/src/main/java/com/example/protego/PatientDashboardActivity.java

nickdlc commented on client/app/src/main/res/layout/activity_patient_general_info.xml:
```diff
@@ -0,0 +1,879 @@
+                        android:layout_width="wrap_content"
+                        android:layout_height="match_parent">
+
+                        <TextView
```
Long addresses are cut off from the general info display (e.g. "401 W 130th St, New York, NY 10027" displays as "401 W 130th St, New").

chelseaL01 replied:
I updated the general information and emergency contact information textviews to show long texts.

chelseaL01 replied:
I updated the onboarding design to show required onboarding fields with an * before the question and with the phrase "(Required)" after the question. We can highlight the required fields in a different way if you think there is better way to show it ?

nickdlc replied:
I think * is just enough and we can maybe just say "* denotes required field" at the top of the page.

nickdlc commented on client/app/src/main/java/com/example/protego/PatientOnboardingActivity.java:
```diff
@@ -2,26 +2,238 @@
+
+                Height_Inches = Height_Inches_edit.getText().toString();
+                Weight = Weight_edit.getText().toString();
+                String Height = Height_Feet + "'" + Height_Inches + " feet";
```
Height is displayed incorrectly (e.g. `5'4"` displays as `5'4 feet`). Alternatives could be `x feet, y inches` or `x'y"`.

chelseaL01 replied:
A patient's height is now saved to firestore with the units "feet" and "inches" and is shown on the general information page.

nickdlc commented on client/app/src/main/java/com/example/protego/PatientGeneralInfoActivity.java:
```diff
@@ -0,0 +1,326 @@
+    public static PatientCancerRecyclerViewAdapter cancer_adapter;
+    public static PatientDiabetesRecyclerViewAdapter diabetes_adapter;
+    public static PatientOtherConditionsRecyclerViewAdapter other_conditions_adapter;
+    public final static String[] phoneData = new String[1];
```
I think it should be possible to avoid instances of having to use final, static arrays of size 1, but this isn't the most important change given the time remaining (There should be some instances where this is avoided somewhere, but I don't recall off the top of my head).

chelseaL01 replied:
I made updates to the PatientDashboard and PatientGeneralInfo Activities. I pull the onboarding fields from the dashboard to display the information in time when the user navigates to PatientGeneralInfo. The variables are now static strings on the dashboard code. Initially I pulled the firestore fields on the general information activity but the UI components would not update in time. I updated this code too to strings if we want to use the code later on

nickdlc reviewed (changes requested):
Addressed a couple minor UI issues. The comment about the final, static arrays of size 1 is more an observation than a request to change things.

Also, I didn't address this in a file directly, but the Onboarding page should denote which fields are required. If I recall correctly, it's everything except medical history, but we should at least denote that so the "Please complete required fields" message is clearer.

chelseaL01 committed:
Update General Information page design + show onboarding required fields

Required onboarding fields appear with an * before the question and with the phrase "(Required)" after the question.

A patient's height is saved to firestore with the units "feet" and "inches" and is shown on the general information page as so.

The general and emergency contact text views are updated to show long texts.

nickdlc committed:
Merge branch 'main' into features/PROT-66-Connect-Onboarding-Fields-to-Firestore

chelseaL01 committed:
Update Patient Dashboard Activity and Patient General Info Activity

I updated the variables to get general and emergency contact information from firestore on the dashboard. Before, these variables were public static String of length 1 now they are public static string. They are static on the Dashboard activity because I pull the data on the dashboard to show on the general information  general information activity.

Initially, I pulled the data from firestore on the general information activity but it would not update the UI components in time to show it to the user. The code to pull data from the general information page is still included in the code to use if we can find a solution to this issue, those variables are not static.

chelseaL01 committed:
Merge branch 'main' of https://github.com/nickdlc/Protego into features/PROT-66-Connect-Onboarding-Fields-to-Firestore

 Conflicts:
	client/.idea/misc.xml

chelseaL01 committed:
Merge branch 'features/PROT-66-Connect-Onboarding-Fields-to-Firestore' of https://github.com/nickdlc/Protego into features/PROT-66-Connect-Onboarding-Fields-to-Firestore

chelseaL01 committed:
Update patient_onboarding.xml

The required fields are shown with an "*" and a message is shown above the form stating that "* Denotes a Required field"
